binaries/cuprated/src/version.rs

```diff
@@ -0,0 +1,73 @@
+//! Misc version information about `cuprated`.
+
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+
+use cuprate_constants::build::{BUILD, COMMIT};
+use cuprate_rpc_types::{CORE_RPC_VERSION, CORE_RPC_VERSION_MAJOR, CORE_RPC_VERSION_MINOR};
+use cuprate_types::HardFork;
+
+use crate::{
+    constants::{MAJOR_VERSION, MINOR_VERSION, PATCH_VERSION, VERSION},
+    killswitch::KILLSWITCH_ACTIVATION_TIMESTAMP,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct CupratedVersionInfo {
+    /// `cuprated`'s major version.
+    major_version: u8,
+    /// `cuprated`'s minor version.
+    minor_version: u8,
+    /// `cuprated`'s patch version.
+    patch_version: u8,
+    /// RPC major version (from `monerod`).
+    rpc_major_version: u32,
+    /// RPC minor version (from `monerod`).
+    rpc_minor_version: u32,
+    /// RPC version (from `monerod`).
+    rpc_version: u32,
+    /// The latest hardfork supported.
+    hardfork: u8,
+    /// The blockchain database version specific to `cuprated`.
+    blockchain_db_version: u64,
+    /// `cuprated`'s semantic version.
+    semantic_version: &'static str,
+    /// Build type, either `debug` or `release`.
+    build: &'static str,
+    /// Git commit hash of the build.
+    commit: &'static str,
+    /// Timestamp at which `cuprated`'s killswitch activates.
+    killswitch_timestamp: u64,
+}
```

Comment:
This replaces the `--version` output for:
1. More info
2. JSON (for easier interopt)

This is added in this PR because we can add killswitch info.

```bash
cargo run --bin cuprated -- --version
```
```json
{
  "major_version": 0,
  "minor_version": 0,
  "patch_version": 1,
  "rpc_major_version": 3,
  "rpc_minor_version": 14,
  "rpc_version": 196622,
  "hardfork": 16,
  "blockchain_db_version": 0,
  "semantic_version": "0.0.1",
  "build": "debug",
  "commit": "fb6ec1710b64c8a822aea227c0140449a681b94a",
  "killswitch_timestamp": 1740805200
}
```